### PR TITLE
Develop paid debt

### PR DESCRIPTION
**Pull request by mrpandat**

### CODE REVIEW

**Comment by mrpandat** [comment]
J'ai try hard pour sortir ce code aujourd'hui, je ferai le lintage plus tard.

**Comment by mrpandat** [comment]
--> Paiement des croissants ✅ 
--> Mise à jour de la doc avec le paiement des croissants ✅ 
--> Finalisation des user blacklisted ✅ 
--> Mise à jour de la doc pour les user blacklisted ✅

File: src/crobot/utils.js [hunk 1]
```diff
@@ -21,3 +21,19 @@ export function throwIfNull(key: string): string {
...
+  for (var uid in users) {
+    if (users[uid].id == id) {
+      return users[uid].name;
+    }
+  }
+}
+
+export function usernameToUserid(username: string, users: any): ?string {
+  for (var uid in users) {
+    if (users[uid].name == username) {
```
Comment (Erazihel) [inline src/crobot/utils.js]: Use `===` instead.

File: src/crobot/utils.js [hunk 2]
```diff
@@ -21,3 +21,19 @@ export function throwIfNull(key: string): string {
 
   return process.env[key];
 }
+
+export function userIdToUsername(id: string, users: any): ?string {
+  for (var uid in users) {
+    if (users[uid].id == id) {
```
Comment (Erazihel) [inline src/crobot/utils.js]: Use `===` instead.

File: src/crobot/index.js [hunk 3]
```diff
@@ -82,13 +92,26 @@ export class Crobot {
 
     if (taggedUsernames.includes(this.name)) {
       let responseMessage;
-
+      let sendUser = message.user;
```
Comment (Erazihel) [inline src/crobot/index.js]: `const {user: sendUser} = message;`

File: src/crobot/index.js [hunk 4]
```diff
@@ -82,13 +92,26 @@ export class Crobot {
...
+        responseMessage = this.onBlacklistList();
       } else if (text.match(/list/g)) {
         responseMessage = getCroissantedList();
+      } else if (text.match(/paid/g)) {
+        responseMessage = this.onVoteForUncroissantedUser(
+          sendUser,
+          taggedUsernames
+        );
+      } else if (text.match(/leviosa/g)) {
+        responseMessage = `Stop it <@${message.user}>, stoooooop :leviosa:`;
```
Comment (Erazihel) [inline src/crobot/index.js]: `sendUser` instead of `message.user`

File: src/crobot/index.js [hunk 5]
```diff
@@ -82,13 +92,26 @@ export class Crobot {
...
+      } else {
+        responseMessage = `<@${
+          message.user
```
Comment (Erazihel) [inline src/crobot/index.js]: `sendUser` instead of `message.user`

File: src/crobot/croissants.js [hunk 6]
```diff
@@ -1,20 +1,97 @@
...
+    return croissantsCount;
+  }
+  return 0;
+}
+
+export function voteForUncroissantedUser(
+  croissantedUser: ?string,
+  userVote: string
+): string {
+  if (!croissantedUser) {
```
Comment (Erazihel) [inline src/crobot/croissants.js]: Let your code breathe!

File: src/crobot/croissants.js [hunk 7]
```diff
@@ -1,20 +1,97 @@
...
+    return `Error`;
+  }
+  if (croissantedUser == userVote) {
+    const newCroissantsCount = increaseUserCroissantsCount(croissantedUser);
+    return `<@${croissantedUser}> You are a real cheater are'nt you ? One more breakfast to bring, you now have a debt of ${newCroissantsCount} more croissants ! :croissant:`;
```
Comment (Erazihel) [inline src/crobot/croissants.js]: `aren't`
Remove the `more` word in the sentence `you now have a debt of ${newCroissantsCount} more croissants`

File: src/crobot/croissants.js [hunk 8]
```diff
@@ -1,20 +1,97 @@
...
+      vote: [],
+    };
+    return croissantsCount;
+  }
+  return 0;
+}
+
+export function voteForUncroissantedUser(
+  croissantedUser: ?string,
+  userVote: string
```
Comment (Erazihel) [inline src/crobot/croissants.js]: Rename this variable to `votingUser`, it's more understandable IMO.

File: src/crobot/croissants.js [hunk 9]
```diff
@@ -1,20 +1,97 @@
...
+
+export function increaseVoteList(croissantedUser: string, userVote: string) {
+  croissantedUsers[croissantedUser].vote.push(userVote);
+}
+
+export function resetVoteList(croissantedUser: string) {
+  croissantedUsers[croissantedUser].vote = [];
+}
+
+export function checkIfUserHasVoted(user: string, userVote: string) {
```
Comment (Erazihel) [inline src/crobot/croissants.js]: Rename this function `hasUserVoted`.

File: src/crobot/croissants.js [hunk 10]
```diff
@@ -1,20 +1,97 @@
...
+  if (
```
Comment (Erazihel) [inline src/crobot/croissants.js]: Directly return the condition in the `if` statement:
```
return croissantedUsers[user] && croissantedUsers[user].vote && croissantedUsers[user].vote.indexOf(userVote) >= 0;
```

File: README.md [hunk 11]
```diff
@@ -50,11 +58,21 @@ This will output something like:
...
+
+To blacklist you:
+> @crobot blacklist me
+
+To get a list of blacklisted users:
+
+> * `<user1>`
+> * `<user2>`
+> * `<user3>`
+
```
Comment (Erazihel) [inline README.md]: Add documentation about how to get the blacklisted users list.

File: src/crobot/blacklist.js [hunk 12]
```diff
@@ -26,3 +26,12 @@ export function blackListUser(user: string, croissantsCount: number): string {
 
   return messages.join('\n');
 }
+
+export function blacklistList(): string {
+  const messages = [];
+  messages.push(`Here's a list of all blacklisted users (shame to them) :`);
+  blackListedUsers.map(blUser => {
```
Comment (Erazihel) [inline src/crobot/blacklist.js]:
```
const messages = [
    `Here's a list of all blacklisted users (shame to them) :`,
    ...blackListedUsers.map(blUser => `<@${blUser}>`)
];

return messages.join('\n');
```

File: src/crobot/croissants.js (hunk 6 above)
Comment (mrpandat) [reply src/crobot/croissants.js]: Nany ?

File: src/crobot/croissants.js [hunk 13]
```diff
@@ -11,30 +11,27 @@ export function getUserCroissantsCount(user: string): number {
   return croissantedUsers[user] ? croissantedUsers[user].debt : 0;
 }
 
-export function getUserVoteCount(croissantedUser: string): number {
+export function getvotingUserCount(croissantedUser: string): number {
```
Comment (Erazihel) [inline src/crobot/croissants.js]: `getVotingUserCount` 😱

**Pull request merged**